MC/config/PWGHF/external/generator/generator_pythia8_gaptriggered_hf.C

```diff
@@ -56,28 +78,46 @@ public:
 
 protected:
   //__________________________________________________________________
-  bool generateEvent() override {
+  bool generateEvent() override
+  {
+    
 
     // Simple straightforward check to alternate generators
-    if (mGeneratedEvents % mInverseTriggerRatio == 0) {
+    if (mGeneratedEvents % mInverseTriggerRatio == 0)
+    {
+      int nInjectedEvents = mGeneratedEvents / mInverseTriggerRatio;
+      // Alternate quarks if enabled (with the same ratio)
+      if (mQuarkPdgList.size() >= 1)
+      {
+        int iQuark = nInjectedEvents % mQuarkPdgList.size();
+        mQuarkPdg = mQuarkPdgList[iQuark];
+        LOG(debug)<<"SELECTED quark: "<<mQuarkPdgList[iQuark];
+      }
+      // Alternate hadrons if enabled (with the same ratio)
+      if (mHadronPdgList.size() >= 1)
+      {
+        int iHadron = (nInjectedEvents / std::max(mHadronPdgList.size(), 1ul)) % mHadronPdgList.size();
```

Comment:
```suggestion
        int iHadron = (nInjectedEvents / std::max(mQuarkPdgList.size(), 1ul)) % mHadronPdgList.size();
```

Reply:
@mpuccio In this way you change hadron when you injected all quarks one time. E.g., if you have 2 quarks (charm and beauty), you change hadron every c-b pair. Right?

Reply:
Yes indeed